pkg/planner/core/logical_plan_builder.go

```diff
@@ -1287,6 +1289,7 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p base.LogicalPlan, wh
 	b.optFlag |= flagPredicatePushDown
 	b.optFlag |= flagDeriveTopNFromWindow
 	b.optFlag |= flagPredicateSimplification
+	b.optFlag |= flagConvertOuterToInnerJoin
```

Comment:
Same as above

Reply:
Same as above as well

Reply:
Actually, for selection, there is a possibility of passing down predicate from selection to joins.